config/routes.rb

```diff
@@ -20,9 +20,13 @@
     resource :vote, controller: 'comments/votes', only: [:create, :destroy]
   end
 
-  get ':username' => 'users/lists#index', as: :user_profile
+  get ':reference_id/:id/remove' => 'links#destroy', as: :destroy_link
+  get ':username/:list_id/:id/edit_abstract' => 'references#edit_abstract', as: :edit_abstract
```

Comment:
These two paths should be made RESTful. The path for a destroy action should be a `DELETE` on the id, and editing a reference should happen on the reference path through an `UPDATE` action